core/blockchain_reader.go

```diff
@@ -266,6 +266,9 @@ func (bc *BlockChain) GetAncestor(hash common.Hash, number, ancestor uint64, max
 // transaction indexing is already finished. The transaction is not existent
 // from the node's perspective.
 func (bc *BlockChain) GetTransactionLookup(hash common.Hash) (*rawdb.LegacyTxLookupEntry, *types.Transaction, error) {
+	bc.txLookupLock.RLock()
+	defer bc.txLookupLock.RUnlock()
```

Comment:
I'm a bit hesitant here. Basically, if we ever run into one of those "very large reorg" scenarios, or something is really slow because "reasons", this will add to the pile-up of things hanging. 

Would it make sense to have some sort of `TryLock` mechanism here, so that if we don't obtain the lock, then fine, we'll just not use the cache right now? 

I'm not sure, maybe this is fine